tests/kernel/mem_protect/syscalls/src/main.c

```diff
@@ -253,6 +254,9 @@ void test_user_string_alloc_copy(void)
 
 	ret = string_alloc_copy("this is a kernel string");
 	zassert_equal(ret, 0, "string should have matched");
+
+	ret = string_alloc_copy(src);
+	zassert_equal(ret, -1, "got %d", ret);
```

Comment:
could the -1 be instead of -EPERM here?

Reply:
the return value is defined in z_vrfy_string_alloc_copy() and z_impl_string_alloc_copy(), so I think it is Inappropriate to use `-EPERM` here.
```
int z_impl_string_alloc_copy(char *src)
{
	if (!strcmp(src, kernel_string)) {
		return 0;
	} else {
		return -2;
	}
}

static inline int z_vrfy_string_alloc_copy(char *src)
{
	char *src_copy;
	int ret;

	src_copy = z_user_string_alloc_copy((char *)src, BUF_SIZE);
	if (!src_copy) {
		return -1;
	}

	ret = z_impl_string_alloc_copy(src_copy);
	k_free(src_copy);

	return ret;
}
```

Reply:
understand, thx!